src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs

```diff
@@ -220,6 +256,21 @@ internal partial class CodeFixService : ICodeFixService
                 }
             }
 
+            if (!copilotDiagnostics.IsEmpty)
+            {
+                var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
+                var spanToDiagnostics = ConvertToMap(text, copilotDiagnostics);
+
+                foreach (var (span, diagnosticList) in spanToDiagnostics)
+                {
+                    await foreach (var collection in StreamCopilotFixesAsync(
+                        document, span, [.. diagnosticList], fallbackOptions, cancellationToken).ConfigureAwait(false))
```

Comment:
why is the diagnostic list being copied?

Reply:
Using the existing pattern in this method. Definitely scope for cleanup in this file.